### ISSUE TITLE
Update ecs-agent package (1.57.0+)

### ISSUE THREAD

**Issue by cbgbt** [bottlerocket-os/bottlerocket#1812]
Attempts to update the version of `ecs-agent` have resulted in a build failure while running the go compiler, with little indication as to the root cause. `ecs-agent 1.55.5` update the golang version to 1.17.2, whereas the bottlerocket SDK uses 1.16. Could be related to the inability to build.

**Comment by cbgbt** [comment]
The following patch was applied to fixup `0002-bottlerocket-remove-unsupported-capabilities.patch`, which doesn't cleanly apply to ecs-agent 1.57.0.

```
diff --git a/packages/ecs-agent/0002-bottlerocket-remove-unsupported-capabilities.patch b/packages/ecs-agent/0002-bottlerocket-remove-unsupported-capabilities.patch
index 2b1c5f21..78328557 100644
--- a/packages/ecs-agent/0002-bottlerocket-remove-unsupported-capabilities.patch
+++ b/packages/ecs-agent/0002-bottlerocket-remove-unsupported-capabilities.patch
@@ -11,7 +11,7 @@ diff --git a/agent/app/agent_capability.go b/agent/app/agent_capability.go
 index e05430c6d..227fc9462 100644
 --- a/agent/app/agent_capability.go
 +++ b/agent/app/agent_capability.go
-@@ -235,33 +235,33 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
+@@ -231,36 +231,36 @@ func (agent *ecsAgent) capabilities() ([]*ecs.Attribute, error) {
        capabilities = agent.appendPIDAndIPCNamespaceSharingCapabilities(capabilities)
  
        // ecs agent version 1.26.0 supports aws-appmesh cni plugin
@@ -34,6 +34,10 @@ index e05430c6d..227fc9462 100644
 -      capabilities = agent.appendFirelensLoggingDriverCapabilities(capabilities)
 +      // capabilities = agent.appendFirelensLoggingDriverCapabilities(capabilities)
  
+       // support aws router capabilities for log driver router config
+-      capabilities = agent.appendFirelensLoggingDriverConfigCapabilities(capabilities)
++      // capabilities = agent.appendFirelensLoggingDriverConfigCapabilities(capabilities)
+ 
        // support efs on ecs capabilities
        capabilities = agent.appendEFSCapabilities(capabilities)
  
@@ -49,7 +53,7 @@ index e05430c6d..227fc9462 100644
 -              capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
 -      }
 +      // for _, cap := range agent.cfg.VolumePluginCapabilities {
-+      //      capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
++      //      capabilities = agent.appendEFSVolumePluginCapabilities(capabilities, cap)
 +      // }
  
        // support fsxWindowsFileServer on ecs capabilities
```

**Comment by samuelkarp** [comment]
> Attempts to update the version of ecs-agent have resulted in a build failure while running the go compiler, with little indication as to the root cause.

Do you have logs from the build failure?

**Comment by cbgbt** [comment]
This has been resolved in #1816
